src/katgpucbf/fgpu/engine.py

```diff
@@ -681,6 +681,7 @@ async def _flush_out(self, new_timestamp: int) -> None:
         # data).
         accs = self._out_item.n_spectra // self.output.spectra_per_heap
         self._out_item.n_spectra = accs * self.output.spectra_per_heap
+        self._out_item.present[self._out_item.n_spectra :] = False
```

Comment:
I worry that this may be expensive in some corner cases (although it is probably fine). It would be cheaper to modify `_update_dig_power_sensors` to check
```python
if out_item.n_spectra == out_item.capacity and np.all(out_item.present):
```

Reply:
True, and that would solve this problem.

But later on we might check for all packets present and get surprised by this again.

Unless we do something like this for the `jjjOutQueueItem`: 

```python
@property 
def all_present(self) -> bool:
    return self.n_spectra == self.capacity and np.all(self.present)
```

Reply:
I like the all_present idea.